packages/terra-form-select/src/shared/isMobile.jsx

```diff
@@ -0,0 +1,16 @@
+/**
+ * @desc Determine if the browser is ran on a mobile device
+ *
+ * @returns true if mobile (includes tablets), else false
+ *
+ * On the userAgent check against "Mobi", quoting MDN:
+ * > In summary, we recommend looking for the string “Mobi” anywhere in the
+ * > User Agent to detect a mobile device
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#Mobile_Tablet_or_Desktop
+ *
+ * We also need to ensure that touch events are present in the window, since our
+ * assumption is that mobile uses touch as the primary interface
+ */
+export default function isMobile() {
+  return /Mobi/.test(navigator.userAgent) && 'ontouchstart' in window;
```

Comment:
before iOS 13, this worked on all mobile devices

however, iPads will no longer say Mobile and will instead use the Desktop experience that mac os provides

surprisingly, Apple did add support for [visual viewport](https://developer.mozilla.org/en-US/docs/Web/API/Visual_Viewport_API), so we can fall back to this

this is an interesting problem to solve because we don't want to simply say "if the device ~quacks like a duck, it must be a duck~ supports touch, then it's mobile" This is because more and more desktop PCs are supporting touch and we really only want to target devices that use a virtual on-screen keyboard.

reworking this since it affects how we handle blurring the input on keyboard overlay devices (phones & tablets)

Reply:
turns out we don't necessarily need this for the blur logic... it should be able to be simplified by attaching `onTouchStart` to the select's `<input>` and setting internal state for something like `inputFocusedByTouch`

since the `<input>` `onBlur` fires when the onscreen keyboard is dismissed, we can now better handle this in the frame `onBlur` by looking at `inputFocusedByTouch`

still need to explore what happens when the user just taps on a new input though... 🧐

however, we should keep the above comment in mind when thinking about our mobile voiceover specific logic

Reply:
so, I can determine when a touch event focuses the select, even with voice over on

using that, I can determine if I should blur the select or not (need to look at the `event.relatedTarget`, which will be `null` if the keyboard was dismissed, else populated by another target

the current behavior of blurring the select closes the menu and clears the input search value

however, by not blurring, the select menu will be visible until interacted with or manually closed...

Reply:
What's this used to solve again?